### DISCUSSION TITLE
Code Block inside a collapsable Admonitions

### DISCUSSION THREAD

**Discussion by gerrysweeney** [squidfunk/mkdocs-material#3398] [Q&A] [answered]
Is it possible to include a syntax coloured code block inside a collapsible Admonitions. This would be quite helpful for me but I cannot seem to find a way of doing it. 

Thanks
Gerry

**Comment by facelessuser** [comment]
Do you mean something like this? https://facelessuser.github.io/pymdown-extensions/extensions/details/#syntax

**Reply by facelessuser** [reply]
They work just like Admonitions, but they are collapsible. You can put code or whatever in them.

**Comment by squidfunk** [comment, marked as answer]
Yes, the documentation even includes several examples, e.g. in the [Docker installation guide](https://squidfunk.github.io/mkdocs-material/getting-started/#with-docker):

<img width="713" alt="Bildschirmfoto 2021-12-30 um 11 31 59" src="https://user-images.githubusercontent.com/932156/147743871-3d23e574-1f3e-42dc-9e74-5568ab25c3c2.png">

Source:

``` markdown
??? question "How to add plugins to the Docker image?"

    Material for MkDocs only bundles selected plugins in order to keep the size
    of the official image small. If the plugin you want to use is not included, 
    create a new `Dockerfile` and extend the official Docker image:

    ``` Dockerfile
    FROM squidfunk/mkdocs-material
    RUN pip install ...
    ```

    Next, you can build the image with the following command:

    ```
    docker build -t squidfunk/mkdocs-material .
    ```

    The new image can be used exactly like the official image.
```

**Reply by FernandoBasso** [reply]
Is it possible to have more content with multiple lines, code blocks, etc. in the question? Something like [this](https://www.devhowto.dev/haskell/hffp/ch04-data-types/qa-data-types.html) (a toy project of mine built with Sphinx + reST, those QA blocks are coded by myself with a dozen lines of CSS and JS, they are not a feature of my Sphinx theme).

![image](https://user-images.githubusercontent.com/6560620/154670323-15116aee-0b4b-458b-b1c4-d4e255ad618e.png)

If not possible at the moment, would it be something worth adding to the the theme, or best left to whoever needs to implement them manually by extending the theme, like I did in the Sphinx case?

**Reply by squidfunk** [reply]
Yes, you can nest arbitrary content inside an admonition, including code blocks and even admonitions themselves.

**Reply by FernandoBasso** [reply]
Sorry, I did not express myself clearly.

Yes, that works fine and looks awesome, but I my question if it is possible to have more content **inside the question**, so that more content is visible in multiple lines inside the question.

![image](https://user-images.githubusercontent.com/6560620/154796171-1b853f8b-316c-41b3-9638-8efbf7fd2d48.png)

**Reply by squidfunk** [reply]
As I said, you can nest admonitions and details:

```
!!! "Question"

    This is visible

    ??? "Answer"

        This is hidden
```

**Reply by FernandoBasso** [reply]
Oh, gosh, I didn't pay enough attention to the word "nest". Thanks!
